ip-whitelist/cmd/resource/resource.go

```diff
@@ -0,0 +1,247 @@
+package resource
+
+import (
+	"context"
+	"encoding/base64"
+	"fmt"
+	"log"
+	"strings"
+
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/encoding"
+	"github.com/aws-cloudformation/cloudformation-cli-go-plugin/cfn/handler"
+	"github.com/mongodb/go-client-mongodb-atlas/mongodbatlas"
+	"github.com/mongodb/mongodbatlas-cloudformation-resources/util"
+)
+
+// Create handles the Create event from the Cloudformation service.
+func Create(req handler.Request, prevModel *Model, currentModel *Model) (handler.ProgressEvent, error) {
+	client, err := util.CreateMongoDBClient(*currentModel.ApiKeys.PublicKey.Value(), *currentModel.ApiKeys.PrivateKey.Value())
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	projectID := *currentModel.ProjectId.Value()
+
+	request := getProjectIPWhitelistRequest(currentModel)
+
+	_, _, err = client.ProjectIPWhitelist.Create(context.Background(), projectID, request)
+	if err != nil {
+		return handler.ProgressEvent{}, err
+	}
+
+	var withelist []string
+	whiteListMap(request, func(entry string) {
+		withelist = append(withelist, entry)
+	})
+
+	id := encodeStateID(map[string]string{
+		"project_id": projectID,
+		"entries":    strings.Join(withelist, ","),
+	})
+
+	currentModel.Id = encoding.NewString(id)
+
+	return handler.ProgressEvent{
+		OperationStatus: handler.Success,
+		Message:         "Create Complete",
+		ResourceModel:   currentModel,
+	}, nil
+}
+
+func getProjectIPWhitelistRequest(model *Model) []*mongodbatlas.ProjectIPWhitelist {
+	var whitelist []*mongodbatlas.ProjectIPWhitelist
+	for _, w := range model.Whitelist {
+		wl := &mongodbatlas.ProjectIPWhitelist{}
+		if w.Comment != nil {
+			wl.Comment = *w.Comment.Value()
+		}
+		if w.CidrBlock != nil {
+			wl.CIDRBlock = *w.CidrBlock.Value()
+		}
+		if w.IpAddress != nil {
+			wl.IPAddress = *w.IpAddress.Value()
+		}
+
+		whitelist = append(whitelist, wl)
+
+	}
+	return whitelist
+}
+
+func whiteListMap(whitelist []*mongodbatlas.ProjectIPWhitelist, f func(string)) {
+	for _, entry := range whitelist {
+		if entry.CIDRBlock != "" {
+			f(entry.CIDRBlock)
+		} else if entry.IPAddress != "" {
+			f(entry.IPAddress)
+		}
+	}
+}
+
+func encodeStateID(values map[string]string) string {
+	encode := func(e string) string { return base64.StdEncoding.EncodeToString([]byte(e)) }
+	encodedValues := make([]string, 0)
+
+	for key, value := range values {
+		encodedValues = append(encodedValues, fmt.Sprintf("%s:%s", encode(key), encode(value)))
+	}
+	return strings.Join(encodedValues, "-")
```

Comment:
If we do add support for AWS SG as above, I think we'll have to change this delimiter